File: backend/src/main/java/com/disaster/backend/service/DisasterService.java
package com.disaster.backend.service;

import com.disaster.backend.request.CreateDisasterRequest;
import com.disaster.backend.response.DisasterResponse;

import java.util.List;

public interface DisasterService {

    DisasterResponse createDisaster(CreateDisasterRequest request);

    List<DisasterResponse> getAllDisasters();

    DisasterResponse getDisasterById(Long id);

    DisasterResponse updateDisaster(Long id, CreateDisasterRequest request);

    void deleteDisaster(Long id);
}

File: frontend/src/services/disasterService.js
import api from "./api";

export const getDisasters = () => api.get("/disasters");

export const createDisaster = (disaster) =>
    api.post("/disasters", disaster);

export const updateDisaster = (id, disaster) =>
    api.put(`/disasters/${id}`, disaster);

export const deleteDisaster = (id) =>
    api.delete(`/disasters/${id}`);